libraries/Microsoft.Bot.StreamingExtensions/Request.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+
+namespace Microsoft.Bot.StreamingExtensions
+{
+    /// <summary>
+    /// Implementation split between Request and RequestEx.
+    /// The basic request type sent over Bot Framework Protocol 3 with Streaming Extensions transports,
+    /// equivalent to HTTP request messages.
+    /// </summary>
+    public partial class Request
+    {
```

Comment:
This class has me so confused. 

We have :
1. Request.
2. RequestEx
3. RequestExtensions
4. ReceiveRequest
5. ReceiveRequestExtensions. 

That's 5 files that should be one. Please rationalize.

Reply:
Request and ReceiveRequest are different, each has a set of extension methods that are helpers. When we did the code review for this in our other branch, the opinion was to keep the extension methods out of the POCO object. RequestEx and RequestExtensions should be consolidated.

Reply:
Changed name to StreamingRequest and rolled RequestEx into Request. Can't combine RequestEx and RequestExtensions without making the result non-static. If we'd rather do that let me know.